### DISCUSSION TITLE
Configuration paramter

### DISCUSSION THREAD

**Discussion by Paultn12** [visose/Robots#237] [Q&A] [answered]
Hi,

I recently threw together a pick and place setup, but kept having errors in the code, usually around an axis being out of range. I dont think one solution is reviewing the orientation of the planes, but I wouldn't think any of this should be a huge issue for the robot to reach as long as I can change the configuration.

I have tried playing around with the Configuration parameter on the Targets node, but it wasn't making loads of sense to me and I can't find any documentation on how this component works.

I am hoping others will be able to help me in understanding the configuration component, or having a look over the attached file and seeing if I am missing something.

The other tool I use for offline programming is ABB Robot Studio, in that program you can see the whole tool path (even if/with errors) and then very easily go in and change the configuration to work. I'm relatively new to 6 axis programming, but Robots and RobotStudio is my base and always keen to learn more.

Many thank you's in advance and many apologies for my script housekeeping.
Paul.

[pickandplace_robots.zip](https://github.com/visose/Robots/files/11925830/pickandplace_robots.zip)

**Comment by visose** [comment, marked as answer]
Hi, you can find some info here: https://github.com/visose/Robots/wiki/Parameters
The configuration parameter will work with cartesian joint motions. Linear motions will always use the same configuration that the robot already had at the start of the motion.
